### DISCUSSION TITLE
StateStorage use generic

### DISCUSSION THREAD

**Discussion by guaijie** [pmndrs/zustand#1446] [Ideas]
export interface StateStorage<T> {
    getItem: (name: string) => T | null;
    setItem: (name: string, value: string) => void | Promise<void>;
    removeItem: (name: string) => void | Promise<void>;
}
It's more universal,
Because we can use data storage tools other than localstorage
Such data storage tools, when deserialized, can return more than just a string type

**Comment by dai-shi** [comment]
Yeah, [jotai](https://github.com/pmndrs/jotai) is designed after zustand, and it uses generic storage. We can do something similar, but I wonder if it's possible without breaking the current behavior and types.
